tests/integration_tests/performance/test_block_performance.py

```diff
@@ -0,0 +1,541 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Performance benchmark for block device emulation."""
+import concurrent
+import json
+import logging
+import os
+from enum import Enum
+import shutil
+import pytest
+
+
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.builder import MicrovmBuilder
+from framework.matrix import TestContext, TestMatrix
+from framework.statistics import core, criteria
+from framework.utils import get_cpu_percent, CmdBuilder, eager_map, \
+    get_cpu_model_name
+import host_tools.drive as drive_tools
+import host_tools.network as net_tools  # pylint: disable=import-error
+import framework.statistics as st
+from integration_tests.performance.block_performance_test_config import \
+    DEBUG, FIO, CONFIG, IOPS_READ_MEAN, CLAT_WRITE_MEAN, IOPS_WRITE_MEAN, \
+    SLAT_READ_MEAN, CPU_UTILIZATION_VMM_SAMPLES_TAG, LAT_READ_MEAN, \
+    LAT_WRITE_MEAN, FIO_OUTPUT, SLAT_WRITE_MEAN, CLAT_READ_MEAN, \
+    BW_READ_MEAN, BW_WRITE_MEAN, TEST_ID, IOPS_WRITE, IOPS_READ, BW_WRITE, \
+    BW_READ, CLAT_READ, CLAT_WRITE, SLAT_READ, SLAT_WRITE, LAT_READ, \
+    LAT_WRITE, IOPS_READ_STDDEV, IOPS_WRITE_STDDEV, BW_READ_STDDEV, \
+    BW_WRITE_STDDEV, SLAT_READ_STDDEV, SLAT_WRITE_STDDEV, CLAT_READ_STDDEV, \
+    CLAT_WRITE_STDDEV, LAT_READ_STDDEV, LAT_WRITE_STDDEV, \
+    CPU_UTILIZATION_HOST, CPU_UTILIZATION_GUEST, \
+    CPU_UTILIZATION_VCPUS_TOTAL, CPU_UTILIZATION_VMM
+
+
+def get_cpu_utilization_measurements():
+    """Cpu utilization measurements."""
+    return [st.consumer.MeasurementDef(CPU_UTILIZATION_HOST, "percentage"),
+            st.consumer.MeasurementDef(CPU_UTILIZATION_GUEST, "percentage")]
+
+
+def get_read_measurements():
+    """Return read-only measurements."""
+    return [st.consumer.MeasurementDef(IOPS_READ, "io/s"),
+            st.consumer.MeasurementDef(BW_READ, "KiB/s"),
+            st.consumer.MeasurementDef(SLAT_READ, "ns"),
+            st.consumer.MeasurementDef(CLAT_READ, "ns"),
+            st.consumer.MeasurementDef(LAT_READ, "ns")]
+
+
+def get_rw_measurements():
+    """Return read/write measurements."""
+    stats = get_read_measurements()
+    stats.extend([
+        st.consumer.MeasurementDef(IOPS_WRITE, "io/s"),
+        st.consumer.MeasurementDef(BW_WRITE, "KiB/s"),
+        st.consumer.MeasurementDef(SLAT_WRITE, "ns"),
+        st.consumer.MeasurementDef(CLAT_WRITE, "ns"),
+        st.consumer.MeasurementDef(LAT_WRITE, "ns"),
+    ])
+    return stats
+
+
+def define_measurements(mode):
+    """Define metrics based on the mode."""
+    ms = get_cpu_utilization_measurements()
+    if mode.endswith("read"):
+        ms.extend(get_read_measurements())
+        return ms
+    ms.extend(get_rw_measurements())
+    return ms
+
+
+def get_cpu_utilization_stats(env_id, fio_id, cpu=None):
```

Comment:
Shouldn't we have a generic function for measuring CPU utilization and feeding the expected values?

I don't really understand why we need to define CPU usage comparisons criteria since these should be universal.